winlogbeat/beater/winlogbeat.go

```diff
@@ -100,7 +108,14 @@ func (eb *Winlogbeat) init(b *beat.Beat) error {
 
 		eb.eventLogs = append(eb.eventLogs, logger)
 	}
-
+	b.OverwritePipelinesCallback = func(esConfig *common.Config) error {
```

Comment:
Based on the name "OverwritePipelinesCallback", is this only called when the pipelines are already installed? If not, then this name is confusing name (but that's a problem for another day). This next comment probably also conflicts with the naming.

I would expect that `winlogbeat.overwrite_pipelines` is honored in all contexts that install pipelines. By not honoring the config option we taking control away from the administrators.

Automation in configuration management systems that calls `setup` unconditionally (with the thinking that it would be an efficient idempotent operation) will trigger extra work for Elasticsearch. Given that the pipelines are all versioned, always overwriting them seems like overkill. Automation / CM would probably then want to implement the "does exist" check before calling setup (which would be a pain).

Reply:
The name is very misleading.

This is only called if you are in the `setup` command.
https://github.com/elastic/beats/blob/e6e65aa92fe355c95789691ebf5a3bcecaf5b4ea/libbeat/cmd/instance/beat.go#L590

If the Callback overwrites the pipelines is up to the callback function.  But in Filebeat, when we are in the setup command we always overwrite the pipelines.

https://github.com/elastic/beats/blob/e6e65aa92fe355c95789691ebf5a3bcecaf5b4ea/filebeat/beater/filebeat.go#L202

So for filebeat, if you run setup you **always** overwrite the pipelines, doesn't matter what `filebeat.overwrite_pipelines` is set to.   `filebeat.overwrite_pipelines` only controls if pipelines are overwritten on the ES connect callback.  Pipelines get loaded when ever you connect ES, if they already exist then the variable `filebeat.overwrite_pipelines` determines if they get overwritten or if the existing one is kept.

I duplicated the filebeat behavior.  I think what you suggest is saner.  Do we want sanity or consistency?

Reply:
Sanity 😆